Tests/SwiftStdlibTests/StringExtensionsTests.swift

```diff
@@ -565,26 +565,44 @@ final class StringExtensionsTests: XCTestCase {
         XCTAssertEqual(url, "it's%20easy%20to%20encode%20strings")
     }
 
-    func testMatches() {
+    let emailPattern = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}"
+
+    func testPatternMatches() {
         XCTAssert("123".matches(pattern: "\\d{3}"))
         XCTAssertFalse("dasda".matches(pattern: "\\d{3}"))
-        XCTAssertFalse("notanemail.com".matches(pattern: "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}"))
-        XCTAssert("email@mail.com".matches(pattern: "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}"))
+        XCTAssertFalse("notanemail.com".matches(pattern: emailPattern))
+        XCTAssert("email@mail.com".matches(pattern: emailPattern))
+    }
+
+    func testRegexMatches() {
+        // swiftlint:disable force_try
+        XCTAssert("123".matches(regex: try! NSRegularExpression(pattern: "\\d{3}")))
+        XCTAssertFalse("dasda".matches(regex: try! NSRegularExpression(pattern: "\\d{3}")))
+        XCTAssertFalse("notanemail.com".matches(regex: try! NSRegularExpression(pattern: emailPattern)))
+        XCTAssert("email@mail.com".matches(regex: try! NSRegularExpression(pattern: emailPattern)))
+        // swiftlint:enable force_try
     }
 
     #if canImport(Foundation)
-    func testRegexMatchOperator() {
+    func testPatternMatchOperator() {
         XCTAssert("123" ~= "\\d{3}")
         XCTAssertFalse("dasda" ~= "\\d{3}")
-        XCTAssertFalse("notanemail.com" ~= "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}")
-        XCTAssert("email@mail.com" ~= "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}")
+        XCTAssertFalse("notanemail.com" ~= emailPattern)
+        XCTAssert("email@mail.com" ~= emailPattern)
         XCTAssert("hat" ~= "[a-z]at")
         XCTAssertFalse("" ~= "[a-z]at")
         XCTAssert("" ~= "[a-z]*")
         XCTAssertFalse("" ~= "[0-9]+")
     }
     #endif
 
+    func testRegexMatchOperator() {
+        // swiftlint:disable:next force_try
```

Comment:
Could we just make the test function throws so we don't need to do any of this force `try!` and lint exclusion...?

Reply:
👍

Reply:
I had no idea you could do that, btw 😄